android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/CloudRepository.java

```diff
@@ -168,13 +168,15 @@ public LexicalModel getAssociatedLexicalModel(@NonNull Context context, String l
 
   // Should be called whenever a new language code starts being managed in order to help signal
   // retrieval of the language code's lexical models.
-  public void invalidateLexicalModelCache(@NonNull Context context) {
+  public void invalidateLexicalModelCache(@NonNull Context context, boolean deleteCacheFile) {
     this.invalidateLexicalCache = true;
 
-    // We should also pre-emptively clear out the old cache file
-    // in case of an app close or crash.
-    File file = CloudDataJsonUtil.getLexicalModelCacheFile(context);
-    file.delete();
+    if (deleteCacheFile) {
+      // We should also pre-emptively clear out the old cache file
+      // in case of an app close or crash.
+      File file = CloudDataJsonUtil.getLexicalModelCacheFile(context);
+      file.delete();
```

Comment:
Is it possible for `getLexicalModelCacheFile()` to return `null`?

Reply:
I suppose it's possible. Added a null check